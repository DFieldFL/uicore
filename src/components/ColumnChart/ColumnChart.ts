const defaultConfig = {
  chart: {
    type: 'column',
    width: 400,
    height: 300
  },
  title: {
    text: ''
  },
  yAxis: {
    title: {
      text: ''
    }
  },
  series: [
    {
      name: '',
      data: [71.5, 106.4, 135.6, 148.5, 216.4, 194.1, 95.6, 54.4],
      showInLegend: false
    }
  ],
  credits: {
    enabled: false
  }
}

const getColumnChartDefaultConfig = () => {
  return defaultConfig
}

export { getColumnChartDefaultConfig }
