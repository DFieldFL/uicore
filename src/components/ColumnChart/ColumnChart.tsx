import React from 'react'
import Highcharts, { Options } from 'highcharts'
import HighchartsReact from 'highcharts-react-official'

interface Props {
  width?: string
  height?: string
  options?: Options
}

function ColumnChart(props: Props) {
  const options = {
    chart: {
      type: 'column',
      width: props.width!,
      height: props.height!
    },
    title: {
      text: ''
    },
    yAxis: {
      title: {
        text: ''
      }
    },
    series: [
      {
        name: '',
        data: [71.5, 106.4, 135.6, 148.5, 216.4, 194.1, 95.6, 54.4],
        showInLegend: false
      }
    ],
    credits: {
      enabled: false
    }
  }
  return <HighchartsReact highcharts={Highcharts} options={props.options! || options} />
}

export { ColumnChart }
