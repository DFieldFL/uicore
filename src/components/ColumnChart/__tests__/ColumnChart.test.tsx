import React from 'react'
import { render, wait } from '@testing-library/react'
import { ColumnChart } from '../ColumnChart'

describe('ColumnChart unit tests', () => {
  test('Render ColumnChart should match snapshot', async () => {
    const { container } = render(<ColumnChart />)
    await wait()
    expect(container).toMatchSnapshot()
  })
})
