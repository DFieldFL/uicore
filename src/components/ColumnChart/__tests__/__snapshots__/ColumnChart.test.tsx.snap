// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnChart unit tests Render ColumnChart should match snapshot 1`] = `
<div>
  <div
    data-highcharts-chart="0"
    style="overflow: hidden;"
  >
    <div
      class="highcharts-container "
      dir="ltr"
      id="highcharts-p5ub6nq-0"
      style="position: relative; overflow: hidden; width: 600px; height: 400px; text-align: left; line-height: normal; z-index: 0;"
    >
      <svg
        class="highcharts-root"
        height="400"
        style="font-family:\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Arial, Helvetica, sans-serif;font-size:12px;"
        version="1.1"
        viewBox="0 0 600 400"
        width="600"
        xmlns="http://www.w3.org/2000/svg"
      >
        <desc>
          Created with Highcharts 7.1.2
        </desc>
        <defs>
          <clippath
            id="highcharts-p5ub6nq-1-"
          >
            <rect
              fill="none"
              height="NaN"
              width="NaN"
              x="0"
              y="0"
            />
          </clippath>
        </defs>
        <rect
          class="highcharts-background"
          fill="#ffffff"
          height="400"
          rx="0"
          ry="0"
          width="600"
          x="0"
          y="0"
        />
        <rect
          class="highcharts-plot-background"
          fill="none"
          height="NaN"
          width="NaN"
          x="NaN"
          y="10"
        />
        <g
          class="highcharts-grid highcharts-xaxis-grid"
          data-z-index="1"
        >
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
          />
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
          />
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
          />
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
          />
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
          />
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
          />
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
          />
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
          />
        </g>
        <g
          class="highcharts-grid highcharts-yaxis-grid"
          data-z-index="1"
        >
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
            stroke="#e6e6e6"
            stroke-width="1"
          />
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
            stroke="#e6e6e6"
            stroke-width="1"
          />
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
            stroke="#e6e6e6"
            stroke-width="1"
          />
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
            stroke="#e6e6e6"
            stroke-width="1"
          />
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
            stroke="#e6e6e6"
            stroke-width="1"
          />
          <path
            class="highcharts-grid-line"
            data-z-index="1"
            fill="none"
            stroke="#e6e6e6"
            stroke-width="1"
          />
        </g>
        <rect
          class="highcharts-plot-border"
          data-z-index="1"
          fill="none"
          height="0"
          width="0"
          x="0"
          y="10"
        />
        <g
          class="highcharts-axis highcharts-xaxis"
          data-z-index="2"
        >
          <path
            class="highcharts-tick"
            d="M 0 0"
            fill="none"
            opacity="1"
            stroke="#ccd6eb"
            stroke-width="1"
          />
          <path
            class="highcharts-tick"
            d="M 0 0"
            fill="none"
            opacity="1"
            stroke="#ccd6eb"
            stroke-width="1"
          />
          <path
            class="highcharts-tick"
            d="M 0 0"
            fill="none"
            opacity="1"
            stroke="#ccd6eb"
            stroke-width="1"
          />
          <path
            class="highcharts-tick"
            d="M 0 0"
            fill="none"
            opacity="1"
            stroke="#ccd6eb"
            stroke-width="1"
          />
          <path
            class="highcharts-tick"
            d="M 0 0"
            fill="none"
            opacity="1"
            stroke="#ccd6eb"
            stroke-width="1"
          />
          <path
            class="highcharts-tick"
            d="M 0 0"
            fill="none"
            opacity="1"
            stroke="#ccd6eb"
            stroke-width="1"
          />
          <path
            class="highcharts-tick"
            d="M 0 0"
            fill="none"
            opacity="1"
            stroke="#ccd6eb"
            stroke-width="1"
          />
          <path
            class="highcharts-tick"
            d="M 0 0"
            fill="none"
            opacity="1"
            stroke="#ccd6eb"
            stroke-width="1"
          />
          <path
            class="highcharts-axis-line"
            d="M 0 0"
            data-z-index="7"
            fill="none"
            stroke="#ccd6eb"
            stroke-width="1"
          />
        </g>
        <g
          class="highcharts-axis highcharts-yaxis"
          data-z-index="2"
        >
          <path
            class="highcharts-axis-line"
            d="M 0 0"
            data-z-index="7"
            fill="none"
          />
        </g>
        <g
          class="highcharts-series-group"
          data-z-index="3"
        >
          <g
            class="highcharts-series highcharts-series-0 highcharts-column-series highcharts-color-0  highcharts-tracker"
            clip-path="url(#highcharts-p5ub6nq-1-)"
            data-z-index="0.1"
            transform="translate(0,10) scale(1 1)"
          />
          <g
            class="highcharts-markers highcharts-series-0 highcharts-column-series highcharts-color-0 "
            data-z-index="0.1"
            transform="translate(0,10) scale(1 1)"
          />
        </g>
        <text
          class="highcharts-title"
          data-z-index="4"
          style="color:#333333;font-size:18px;fill:#333333;"
          text-anchor="middle"
          x="300"
          y="24"
        />
        <text
          class="highcharts-subtitle"
          data-z-index="4"
          style="color:#666666;fill:#666666;"
          text-anchor="middle"
          x="300"
          y="10"
        />
        <g
          class="highcharts-legend"
          data-z-index="7"
        >
          <rect
            class="highcharts-legend-box"
            fill="none"
            height="8"
            rx="0"
            ry="0"
            visibility="hidden"
            width="8"
            x="0"
            y="0"
          />
          <g
            data-z-index="1"
          >
            <g />
          </g>
        </g>
        <g
          class="highcharts-axis-labels highcharts-xaxis-labels"
          data-z-index="7"
        >
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0)"
            x="0"
          >
            0
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0)"
            x="0"
          >
            1
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0)"
            x="0"
          >
            2
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0)"
            x="0"
          >
            3
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0)"
            x="0"
          >
            4
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0)"
            x="0"
          >
            5
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0)"
            x="0"
          >
            6
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0)"
            x="0"
          >
            7
          </text>
        </g>
        <g
          class="highcharts-axis-labels highcharts-yaxis-labels"
          data-z-index="7"
        >
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            0
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            50
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            100
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            150
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            200
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            250
          </text>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
